src/liboslexec/optexture.cpp

```diff
@@ -189,34 +219,37 @@ osl_texture_set_interp(void* opt, ustringhash_pod modename_)
         ((TextureOpt*)opt)->interpmode = (TextureOpt::InterpMode)mode;
 }
 
-OSL_SHADEOP void
+OSL_SHADEOP OSL_HOSTDEVICE void
 osl_texture_set_interp_code(void* opt, int mode)
 {
     ((TextureOpt*)opt)->interpmode = (TextureOpt::InterpMode)mode;
 }
 
-OSL_SHADEOP void
+OSL_SHADEOP OSL_HOSTDEVICE void
 osl_texture_set_subimage(void* opt, int subimage)
 {
     ((TextureOpt*)opt)->subimage = subimage;
 }
 
 
-OSL_SHADEOP void
+OSL_SHADEOP OSL_HOSTDEVICE void
 osl_texture_set_subimagename(void* opt, ustringhash_pod subimagename_)
 {
+    // TODO: Enable when subimagename is ustringhash.
+#ifndef __CUDA_ARCH__
     ustringhash subimagename_hash    = ustringhash_from(subimagename_);
     ustring subimagename             = ustring_from(subimagename_hash);
     ((TextureOpt*)opt)->subimagename = subimagename;
```

Comment:
NOT FINAL:  Add hack to pass hash value as ustring to reuse OIIO::TextureOpt vs. creating our own.